src/components/addbook/AddBook.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { addABook } from '../../actions/bookActions';
+import { useNavigate } from 'react-router-dom';
+import './AddBook.css';
+
+const AddBook = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const { userId } = useSelector((state) => state.user);
+  const { isAdded, error } = useSelector((state) => state.books);
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [message, setMessage] = useState('');
+
+  const initialFormData = {
+    title: '',
+    author: '',
+    genre: '',
+    description: '',
+    cover_image_url: '',
+    rental_price: '',
+    available_for_rent: false,
+    condition: '',
+    user_id: userId,
+  };
+
+  const [formData, setFormData] = useState(initialFormData);
+
+  const handleChange = event => {
+    const { name, value } = event.target;
+    setFormData({ ...formData, [name]: value });
+  };
+
+  const handleSubmitBook = (e) => {
+    e.preventDefault();
+    setIsSubmitted(true);
+    dispatch(addABook(formData));
+  };
+
+  useEffect(() => {
+    if (isAdded && isSubmitted) {
+      setMessage('Book is Successfully Added');
+      navigate('/addbook');
```

Comment:
After adding the book, it should navigate to the main page

Reply:
I did that but I felt a user may want to add about 3 books. To avoid back and forth. I left it so the user can access the booklist from books link. Unless is a requirement. 😄